File: Server (ASP.NET Core)/Petadmin.Core/Models/FollowUpAllItem.cs
using PetAdmin.Core.Interfaces;
using PetAdmin.Core.Models;

namespace Petadmin.Core.Models
{
    public class FollowUpAllItem : IGenericDbEntity
    {
        /// <summary>
        /// Keeps the followUp details.
        /// </summary>
        public FollowUp FollowUp { get; set; }

        /// <summary>
        /// Keeps the animal details.
        /// </summary>
        public Animal Animal { get; set; }

        /// <summary>
        /// Keeps the owner details.
        /// </summary>
        public Owner Owner { get; set; }
    }
}

File: Angular Client/models/followup-all-item.ts
import {Followup, FollowupEntity} from './followup';
import {Animal, AnimalEntity} from './animal';
import {Owner, OwnerEntity} from './owner';

export interface FollowupAllItem {
  followup: Followup;
  animal: Animal;
  owner: Owner;
}

export class FollowupAllItemEntity implements FollowupAllItem{
  constructor() {
    this.followup = new FollowupEntity();
    this.animal = new AnimalEntity();
    this.owner = new OwnerEntity();
  }
  followup: Followup;
  animal: Animal;
  owner: Owner;
}
